src/backend/distributed/test/metadata_sync.c

```diff
@@ -80,8 +80,36 @@ wait_until_metadata_sync(PG_FUNCTION_ARGS)
 	uint32 timeout = PG_GETARG_UINT32(0);
 	int waitResult = 0;
 
-	MultiConnection *connection = GetNodeConnection(FORCE_NEW_CONNECTION,
-													"localhost", PostPortNumber);
+	List *workerList = ActivePrimaryNodeList(NoLock);
```

Comment:
@pykello FYI on the change here. Let me know if there is anything that doesn't make sense to you.